### PR TITLE
[develop] Add mysql_repo resource

### PR DESCRIPTION
**Pull request by NSsirena**
### Description of changes
Add custom resource `mysql_repo` to configure mysql repository.
A custom resource was necessary since mysql provides a package (rpm/deb) that when installed configure the required repositories.
This may be useful for security patches and updates.

### Tests
* Resource tested locally on docker on all platforms.
  * Ubuntu20 is failing of a non reachable ip that seems unrelated to this change.

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-slurm/recipes/install.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 dns_domain "Install dns related packages"
 mysql_client 'Install mysql client'
+mysql_repo 'Configure mysql repository'
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-slurm/recipes/install.rb:20]: should we configure the repo before installing mysql client?

Comment (NSsirena) [reply cookbooks/aws-parallelcluster-slurm/recipes/install.rb:20]: This was a design decision to keep control of the installed version and at the same time have the repository configured for taking future (security) updates if needed.

File: cookbooks/aws-parallelcluster-slurm/test/controls/mysql_repo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+control 'mysql_repo_added' do
```
Comment (enrico-usai) [inline cookbooks/aws-parallelcluster-slurm/test/controls/mysql_repo_spec.rb]: should we add `tag:install` as prefix to permit this test to run on CI/CD too?

Comment (NSsirena) [reply cookbooks/aws-parallelcluster-slurm/test/controls/mysql_repo_spec.rb]: Good catch!
Yes we should. I'll update the PR.

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**